File: 231. Power of Two/Approach03.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    bool isPowerOfTwo(int n) {
        if (n <= 0) {
            return false;
        }
        int cnt = 0;
        for(int i=0; i<=30; i++){
            if(n&1){
                cnt++;
            }
            n>>=1;
        }
        return cnt==1?true:false;
    }
};

File: 231. Power of Two/Approach03.py
from typing import *

class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        if(n<=0):
            return False
        cnt = 0
        for i in range(31):
            if(n&1):
                cnt += 1
            n >>= 1
        return True if(cnt==1) else False
        